library/ssl_tls.c

```diff
@@ -2422,6 +2422,14 @@ int mbedtls_ssl_fetch_input( mbedtls_ssl_context *ssl, size_t nb_want )
             if( ret < 0 )
                 return( ret );
 
+            if ( (size_t)ret > len )
```

Comment:
In case `INT_MAX` is larger than `SIZE_MAX`, we should also error out if `ret > SIZE_MAX`.

Reply:
Looking back, [I don't think we should have added code to handle `INT_MAX > SIZE_MAX`](https://github.com/ARMmbed/mbedtls/issues/3577). [It caused warnings with some compilers](https://github.com/ARMmbed/mbedtls/pull/3574)